### PR TITLE
Feat add edit event component

### PR DESCRIPTION
**Pull request by Chiranjibichapagain**

### CODE REVIEW

File: client/src/components/EventUpdateForm/index.tsx [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+  const convertDate = (date: any) => {
+    const convertedDate = date.substr(0, 10)
+    return convertedDate
+  }
+  const date = data ? convertDate(data.date) : ''
+  const expires_at = data ? convertDate(data.expires_at) : ''
+
+  const [fields, handleFields] = useFormFields({
+    title: data ? data.title : '',
+    category: data ? data.category : '',
```
Comment (Rostislav013) [inline client/src/components/EventUpdateForm/index.tsx:25]: is `data ? data.title` same as `data?.title` ? Try to use it shorter way

File: client/src/components/EventUpdateForm/index.tsx [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+    date: date,
+    time: data ? data.time : '',
+    description: data ? data.description : '',
+    maxParticipants: data ? data.max_participants : '',
+    created_by: user,
+    image: data ? data.image : '',
```
Comment (Rostislav013) [inline client/src/components/EventUpdateForm/index.tsx:31]: Try `data?.keysOfData`

File: client/src/components/EventUpdateForm/index.tsx [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+    expires_at: expires_at
+  })
+
+  const handleSubmit = async (event: any) => {
+    event.preventDefault()
+    try {
+      const data = { ...fields, address }
+      await axios({
```
Comment (Rostislav013) [inline client/src/components/EventUpdateForm/index.tsx:39]: Did u have a problem with updating of event cuz of address?

**Review by Rostislav013 — APPROVED** [review]

File: client/src/components/EventUpdateForm/index.tsx (hunk 1 above)
Comment (Chiranjibichapagain) [reply client/src/components/EventUpdateForm/index.tsx:25]: I also need to accommodate else ' ' also there. So how's it possible?

File: client/src/components/EventUpdateForm/index.tsx (hunk 3 above)
Comment (Chiranjibichapagain) [reply client/src/components/EventUpdateForm/index.tsx:39]: yep. without passing address it won't update now.

File: client/src/components/EventUpdateForm/index.tsx (hunk 1 above)
Comment (MikeVedsted) [reply client/src/components/EventUpdateForm/index.tsx:25]: This is a ternary not an optional.
Can also be achieved with e.g. `title: data.title || ''`

File: client/src/Routes.tsx [hunk 4]
```diff
@@ -7,11 +7,17 @@ import ProfilePage from './pages/ProfilePage'
...
 
 const Routes = () => (
   <Switch>
     <ProtectedRoute exact path='/' component={HomePage} />
     <ProtectedRoute exact path='/user/:userId' component={ProfilePage} />
+    <ProtectedRoute
+      exact
+      path='/eventEdit/:eventId'
+      component={EventEditPage}
+    />
```
Comment (MikeVedsted) [inline client/src/Routes.tsx:20]: please change the path to `/event/:eventId/edit` to align with the other paths. 
As a general improvement suggestion, would it maybe be better to have it as part of the MyEvents page, like the create event?

File: client/src/components/EventUpdateForm/index.tsx [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+  const expires_at = data ? convertDate(data.expires_at) : ''
+
+  const [fields, handleFields] = useFormFields({
+    title: data ? data.title : '',
+    category: data ? data.category : '',
+    date: date,
+    time: data ? data.time : '',
+    description: data ? data.description : '',
+    maxParticipants: data ? data.max_participants : '',
+    created_by: user,
```
Comment (MikeVedsted) [inline client/src/components/EventUpdateForm/index.tsx:30]: created_by is not needed. This should not be an option to change.

File: client/src/components/EventUpdateForm/index.tsx [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+  const date = data ? convertDate(data.date) : ''
+  const expires_at = data ? convertDate(data.expires_at) : ''
+
+  const [fields, handleFields] = useFormFields({
+    title: data ? data.title : '',
+    category: data ? data.category : '',
+    date: date,
+    time: data ? data.time : '',
+    description: data ? data.description : '',
+    maxParticipants: data ? data.max_participants : '',
```
Comment (MikeVedsted) [inline client/src/components/EventUpdateForm/index.tsx:29]: please rename maxParticipant to max_participants. It is currently not updating due to the wrong name

File: client/src/components/EventUpdateForm/index.tsx (hunk 3 above)
Comment (MikeVedsted) [reply client/src/components/EventUpdateForm/index.tsx:39]: My suggestion should fix this for both of you 😃

File: client/src/components/EventUpdateForm/index.tsx [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+    event.preventDefault()
+    try {
+      const data = { ...fields, address }
+      await axios({
+        method: 'PUT',
+        url: `/api/v1/events/${eventId}`,
+        data: data
+      })
+      setError('')
+      alert(`Success! Event: ${data.event_id} was successfully updated`)
```
Comment (MikeVedsted) [inline client/src/components/EventUpdateForm/index.tsx:45]: could we add a history.push here?

if you do 

```
setError('')
history.push('/')
alert(`Success! Event: ${data.event_id} was successfully updated`)
```

the user will be redirected after confirming.

File: client/src/pages/EventEditPage/index.tsx [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+        setLoading(true)
+        const { data } = await axios.get(`/api/v1/events/${eventId}`)
+        setData(data)
+        setLoading(false)
+      } catch (error) {
+        console.log(error)
+      }
+    }
+    getEventInfo()
+  }, [eventId])
```
Comment (MikeVedsted) [inline client/src/pages/EventEditPage/index.tsx:29]: move to a hook?

**Review by MikeVedsted — APPROVED** [review]

**Comment by MikeVedsted** [comment]
Ideally I would have liked to just use the same form as for creating events but if that is out of scope please create an issue to refactor this later

**Comment by Chiranjibichapagain** [comment]
> Ideally I would have liked to just use the same form as for creating events but if that is out of scope please create an issue to refactor this later

I thought about it and as we need to use a different title and buttons text and functions, I could not think of an idea for executing that. But if that is better we can do that

**Pull request merged**